smack-core/src/main/java/org/jivesoftware/smack/proxy/ProxySocketConnection.java

```diff
@@ -34,4 +36,17 @@
     void connect(Socket socket, String host, int port, int timeout)
                     throws IOException;
 
+
+    static Function<ProxySocketConnection, ProxyInfo> forProxyType(ProxyInfo.ProxyType proxyType) {
+        switch (proxyType) {
+            case HTTP:
+                return HTTPProxySocketConnection::new;
+            case SOCKS4:
+                return Socks4ProxySocketConnection::new;
+            case SOCKS5:
+                return Socks5ProxySocketConnection::new;
+            default:
+                throw new IllegalStateException();
```

Comment:
If anything this should be an `AssertionError("Unknown proxy type: " + proxyType)`. But since you cover all cases of the`ProxyInfo.ProxyType` enum, do you need a default case at all?

Reply:
Ah, I just move the IllegalStateException part from the other constructor as is, Ill make the change to an Assertion error. 

> do you need a default case at all?

Its just considered easier to understand exhaustive cases without going into the enum declaration with the explicit exception.